spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/health/HealthEndpointAutoConfiguration.java

```diff
@@ -35,18 +35,13 @@
 @Configuration(proxyBeanMethods = false)
 @ConditionalOnAvailableEndpoint(endpoint = HealthEndpoint.class)
 @EnableConfigurationProperties
-@Import({ LegacyHealthEndpointAdaptersConfiguration.class, LegacyHealthEndpointCompatibilityConfiguration.class,
-		HealthEndpointConfiguration.class, ReactiveHealthEndpointConfiguration.class,
+@Import({ HealthEndpointConfiguration.class, ReactiveHealthEndpointConfiguration.class,
 		HealthEndpointWebExtensionConfiguration.class, HealthEndpointReactiveWebExtensionConfiguration.class })
 public class HealthEndpointAutoConfiguration {
 
 	@Bean
-	@SuppressWarnings("deprecation")
-	HealthEndpointProperties healthEndpointProperties(HealthIndicatorProperties healthIndicatorProperties) {
-		HealthEndpointProperties healthEndpointProperties = new HealthEndpointProperties();
-		healthEndpointProperties.getStatus().getOrder().addAll(healthIndicatorProperties.getOrder());
-		healthEndpointProperties.getStatus().getHttpMapping().putAll(healthIndicatorProperties.getHttpMapping());
-		return healthEndpointProperties;
+	HealthEndpointProperties healthEndpointProperties() {
```

Comment:
Now that this object can be created using the default constructor, I think the whole bean definition should go away and the class-level annotation should be changed to `@EnableConfigurationProperties(HealthEndpointProperties.class)`.